guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,581 @@
+
+import Ember from 'ember';
+import ToriiAuthenticator from 'ember-simple-auth/authenticators/torii';
+
+export default ToriiAuthenticator.extend({
+  torii: Ember.inject.service()
+});
+```
+
+We also need to define a Torii provider. Because Torii doesn't provide a generator, we need to create the `app/torii-providers`
+directory ourselves and create the following `github.js` inside it.
+
+```js
+// app/torii-providers/github.js
+
+import GitHubOAuth2Provider from 'torii/providers/github-oauth2';
+
+export default GitHubOAuth2Provider.extend({
+  fetch(data) {
+    return data;
+  }
+});
+```
+
+There's only one more piece in this step, connecting our "Log in" button to the authentication mechanism. First, add an
+action to your `login` route.
+
+```js
+  // add to app/routes/login.js
+  
+  actions: {
+    login() {
+      this.get('session').authenticate('authenticator:torii', 'github');
+    }
+  }
+```
+
+Finally, change your `login` template to send the action when the button is pressed.
+
+```handlebars
+{{!-- app/templates/login.hbs --}}
+
+<button {{action "login"}}>Log in to GitHub</button>
+```
+
+We've now established the mechanism to obtain an authorization code from GitHub. This doesn't authorize us fully to use
+the GitHub APIs, although `ember-simple-auth` considers us authenticated at this point, but we can add some information
+to our `application` template that at least verifies things work so far. Add the following line to your `application.hbs`
+after the `API Key` line. When you attempt to login, you should see a popup asking your permission for the app to access
+your account and it will show the authorization code. It will only ask your permission the first time. After that, the
+popup will flash quickly unless you revoke the application's access through GitHub.
+
+```handlebars
+Authorization Code: {{session.data.authenticated.authorizationCode}} <br />
+```
+
+### Logging Out
+
+Before we finish the authentication, let's add the ability to log out. This will help us reset our application so we can
+test it as we go.
+
+First, we'll add a logout button to our `application` template right after the text "Authenticated". This seems a
+reasonable place since it serves as our application's rudimentary menu bar and should only show when the user is
+authenticated.
+
+```handlebars
+{{!-- add to app/templates/application.hbs --}}
+
+    <button {{action "logout"}}>Log Out</button>
+```
+
+Next, add the `logout` action to your `application` controller.
+
+```js
+// add to app/controllers/application.js
+
+  actions: {
+    logout() {
+      this.get('session').invalidate();
+    }
+  }
+```
+
+Clicking the "Log Out" button will take you back to the login state.
+
+### Obtaining an Access Token
+
+The final step in the process is to exchange your authorization code for an API access token. This can only be done
+securely through a back end service because it needs to know your application password. Right now, our app looks like
```

Comment:
I think we should use the correct term _"Client Secret"_ here instead of _"application password"_.